app/rtc_s1.py

```diff
@@ -636,80 +638,94 @@ def run(cfg: RunConfig):
-        geo_obj.orbit = orbit
-        geo_obj.ellipsoid = ellipsoid
-        geo_obj.doppler = zero_doppler
-        geo_obj.threshold_geo2rdr = threshold
-        geo_obj.numiter_geo2rdr = maxiter
+            # init geocode members
+            geo_obj.orbit = orbit
+            geo_obj.ellipsoid = ellipsoid
+            geo_obj.doppler = zero_doppler
+            geo_obj.threshold_geo2rdr = threshold
+            geo_obj.numiter_geo2rdr = maxiter
 
-        # set data interpolator based on the geocode algorithm
-        if geocode_algorithm == isce3.geocode.GeocodeOutputMode.INTERP:
-            geo_obj.data_interpolator = geocode_algorithm
+            # set data interpolator based on the geocode algorithm
+            if geocode_algorithm == isce3.geocode.GeocodeOutputMode.INTERP:
+                geo_obj.data_interpolator = geocode_algorithm
 
-        geo_obj.geogrid(geogrid.start_x, geogrid.start_y,
-                        geogrid.spacing_x, geogrid.spacing_y,
-                        geogrid.width, geogrid.length, geogrid.epsg)
+            geo_obj.geogrid(geogrid.start_x, geogrid.start_y,
+                            geogrid.spacing_x, geogrid.spacing_y,
+                            geogrid.width, geogrid.length, geogrid.epsg)
 
         if save_nlooks:
             nlooks_file = (f'{bursts_output_dir}/{product_prefix}'
                            f'_nlooks.{imagery_extension}')
+
+            if skip_burst_process:
+                os.rename(nlooks_file.replace(output_dir, scratch_path), nlooks_file)
+
             if flag_bursts_secondary_files_are_temporary:
                 temp_files_list.append(nlooks_file)
             else:
                 output_file_list.append(nlooks_file)
-            out_geo_nlooks_obj = isce3.io.Raster(
-                nlooks_file, geogrid.width, geogrid.length, 1,
-                gdal.GDT_Float32, output_raster_format)
+
+            if not skip_burst_process:
+                out_geo_nlooks_obj = isce3.io.Raster(
+                    nlooks_file, geogrid.width, geogrid.length, 1,
+                    gdal.GDT_Float32, output_raster_format)
         else:
             nlooks_file = None
             out_geo_nlooks_obj = None
 
         if save_rtc_anf:
             rtc_anf_file = (f'{bursts_output_dir}/{product_prefix}'
                f'_rtc_anf.{imagery_extension}')
+            # TODO: The if statement below needs to be revised to if- else statement
+            if skip_burst_process:
+                os.rename(rtc_anf_file.replace(output_dir, scratch_path), rtc_anf_file)
+
             if flag_bursts_secondary_files_are_temporary:
                 temp_files_list.append(rtc_anf_file)
             else:
                 output_file_list.append(rtc_anf_file)
-            out_geo_rtc_obj = isce3.io.Raster(
-                rtc_anf_file,
-                geogrid.width, geogrid.length, 1,
-                gdal.GDT_Float32, output_raster_format)
+
+            if not skip_burst_process:
```

Comment:
You use this `skip_burst_process` everywhere in this function. I have a feeling that it's better to create a different function for the case in which `skip_burst_process` is `False`.